mp/src/game/client/neo/ui/neo_hud_friendly_marker.cpp

```diff
@@ -107,49 +109,67 @@ void CNEOHud_FriendlyMarker::DrawNeoHudElement()
 {
 	int x, y;
 	static const float heightOffset = 48.0f;
-	auto pPos = player->GetAbsOrigin();
-	auto pos = Vector(
-		pPos.x,
-		pPos.y,
-		pPos.z + heightOffset);
+	auto pos = player->EyePosition();
 
 	if (GetVectorInScreenSpace(pos, x, y))
 	{
-		auto n = dynamic_cast<C_NEO_Player*>(player);
-		auto a = n->m_rvFriendlyPlayerPositions;
-		static const int maxNameLenght = 32 + 1;		
-		auto playerName = n->GetNeoPlayerName();
-
-		wchar_t playerNameUnicode[maxNameLenght];
-		char playerNameTrimmed[maxNameLenght];
-		snprintf(playerNameTrimmed, maxNameLenght, "%s", playerName);
-		auto textLen = V_strlen(playerNameTrimmed);
-		g_pVGuiLocalize->ConvertANSIToUnicode(playerNameTrimmed, playerNameUnicode, sizeof(playerNameUnicode));
-
-		auto fadeTextMultiplier = GetFadeValueTowardsScreenCentre(x, y);
-		if(fadeTextMultiplier > 0.001)
+		auto playerName = player->GetNeoPlayerName();
+
+		const float fadeTextMultiplier = GetFadeValueTowardsScreenCentre(x, y);
+		if (fadeTextMultiplier > 0.001f)
 		{
+			static constexpr int MAX_MARKER_STRLEN = 48 + 1;
+			const bool localPlayerAlive = const_cast<C_NEO_Player *>(localPlayer)->IsAlive();
+			const bool localPlayerSpec = (localPlayer->GetTeamNumber() < FIRST_GAME_TEAM);
+
 			surface()->DrawSetTextFont(m_hFont);
 			surface()->DrawSetTextColor(FadeColour(teamColor, fadeTextMultiplier));
-			int textWidth, textHeight;
-			surface()->GetTextSize(m_hFont, playerNameUnicode, textWidth, textHeight);
-			surface()->DrawSetTextPos(x - (textWidth / 2), y + m_iMarkerHeight);
-			surface()->DrawPrintText(playerNameUnicode, textLen);
+			int textYOffset = 0;
+			char textASCII[MAX_MARKER_STRLEN];
+
+			auto DisplayText = [this, &textYOffset, x, y](const char *textASCII) {
+				wchar_t textUTF[MAX_MARKER_STRLEN];
+				g_pVGuiLocalize->ConvertANSIToUnicode(textASCII, textUTF, sizeof(textUTF));
+				int textWidth, textHeight;
+				surface()->GetTextSize(m_hFont, textUTF, textWidth, textHeight);
+				surface()->DrawSetTextPos(x - (textWidth / 2), y + m_iMarkerHeight + textYOffset);
+				surface()->DrawPrintText(textUTF, V_strlen(textASCII));
+				textYOffset += textHeight;
+			};
+
+			// Draw player's name and health
+			snprintf(textASCII, MAX_MARKER_STRLEN, "%s", playerName);
+			DisplayText(textASCII);
+
+			// Draw distance to player - Only if local player alive and same team
+			// OR local not alive or spectator and this player has ghost
+			if ((localPlayerAlive && localPlayer->GetTeamNumber() == player->GetTeamNumber()) ||
+					((!localPlayerAlive || localPlayerSpec) && player->IsCarryingGhost()))
+			{
+				const float flDistance = METERS_PER_INCH * player->GetAbsOrigin().DistTo(localPlayer->GetAbsOrigin());
+				snprintf(textASCII, MAX_MARKER_STRLEN, "%s: %.0fm",
+						 player->IsCarryingGhost() ? "GHOST DISTANCE" : "DISTANCE",
+						 flDistance);
+				DisplayText(textASCII);
+			}
+
+			snprintf(textASCII, MAX_MARKER_STRLEN, "%d%%", player->GetHealth());
```

Comment:
Perhaps call `V_snprintf`, so we're letting the game engine figure it out instead of relying on platform specific stdio.h directly.